### ISSUE TITLE
m17-mod-gui (playback) Stream overflow detected!

### ISSUE THREAD

**Issue by lwvmobile** [M17-Project/m17-tools#18]
I've been getting a lot of the following in the console output when running m17-mod-gui in Linux Mint 20 and Linux Mint 21 when PTT is pressed, and really choppy audio output on the modulated 'sound' playing out the speaker. Occasionally, it clears up for a little bit, but most times, its just choppy and I get this message in the terminal.

```
RtApiAlsa::callbackEvent:  audio write error, underrun.
(playback) Stream overflow detected!
```

I did have better luck running this instead to get a modulated output that I could sync up with:

`sox brain.wav -t raw - |  ./m17-mod -S AB1CD -D AB2CD -r | play -q -b 16 -r 48000 -c1 -t s16 -`

Also, a question about the LICH counter, during stream, are the counter values incremented sequentially, or are they broadcast out of order, the pattern I am receiving is 0,3,2,5,4,1. After de-interleaving and Golay24 (which is passing on most things when sync is  lined up) I get the lich counter in that order. Just wanted to verify if that was correct, or if I need to go back and check my de-interleave and bit order endianness, etc.

```
14:33:01 Sync: +M17 STR LC: 0 LICH: [A3][1C][BA][94][07][04]
14:33:01 Sync: +M17 STR LC: 3 LICH: [EF][4D][BF][91][98][64]
14:33:01 Sync: +M17 STR LC: 2 LICH: [32][1C][BA][94][98][44]
14:33:01 Sync: +M17 STR LC: 5 LICH: [32][1C][BA][94][98][A4]
14:33:01 Sync: +M17 STR LC: 4 LICH: [32][1C][BA][FF][4E][84]
14:33:01 Sync: +M17 STR LC: 1 LICH: [32][1C][BA][0B][7B][24]
```

**Comment by Paulo-D2000** [comment]
Hi! Thanks for the bug report! Could you please test the [latest release](https://github.com/M17-Project/m17-tools/releases/tag/v1.1.4) and report if the issue persits ? Thanks.

**Comment by lwvmobile** [comment]
No, I'm afraid the behavior is still present after doing a git pull and rebuilding. Occasionally, it will run just fine, but more often than not, you get a wild stuttering and overflow. Here is a quick video demonstration. WARNING: Turn your volume down.

https://www.youtube.com/watch?v=lNOofb_UbE4

**Comment by Paulo-D2000** [comment]
Strange... it seems like you're running on a VM maybe try it on the main pc ? It should build and work on windows 10/11 too (My dev env. is on win11 msvc 2019...)

**Comment by lwvmobile** [comment]
Actually, running in on real hardware didn't make a difference (again, with Mint) I think I've come to an understanding with it though on getting it to behave properly. If it has an audio feed to modulate before hitting the PTT button, it doesn't seem to have the overflow issue and stabilizes. Not sure why that is, but its good enough to get some testing done. So, I just play it some youtube playlists and hit the ptt button so I can test features in my own software.

**Comment by netmate1234** [comment]
Yes I can confirm this:
RtApiAlsa::callbackEvent:  audio write error, underrun.
(playback) Stream overflow detected!

**Comment by netmate1234** [comment]
On x86 and on arm

**Comment by netmate1234** [comment]
Ubuntu 24 on x86 and Ubuntu 22 for armbian.
